NuKeeper.BitBucket/BitbucketPlatform.cs

```diff
@@ -117,5 +111,14 @@ public Task<SearchCodeResult> Search(SearchCodeRequest search)
         {
             throw new NotImplementedException();
         }
+
+        private static Repository MapRepository(BitBucket.Models.Repository repo)
+        {
+            return new Repository(repo.name, false,
+                    new UserPermissions(true, true, true),
+                    new Uri(repo.links.clone.First().href),
```

Comment:
I just moved this code into a method to avoid repeating it. 

The two links are actually different things so I feel a bit icky moving that into a variable, if you could let me know how the two urls are used, maybe I could actually make them different. 

a) HtmlUrl  - is this a browser url that links you to the repo homepage?
b) CloneUrl - assuming this is  the https (or ssh) clone url

Bitbucket actually sends back the following urls: 

```json
 "links": {
                "watchers": {
                    "href": "https://api.bitbucket.org/2.0/repositories/owner-slug/repostiory-slug/watchers"
                },
                "branches": {
                    "href": "https://api.bitbucket.org/2.0/repositories/owner-slug/repostiory-slug/refs/branches"
                },
                "tags": {
                    "href": "https://api.bitbucket.org/2.0/repositories/owner-slug/repostiory-slug/refs/tags"
                },
                "commits": {
                    "href": "https://api.bitbucket.org/2.0/repositories/owner-slug/repostiory-slug/commits"
                },
                "clone": [
                    {
                        "href": "https://username@bitbucket.org/owner-slug/repostiory-slug.git",
                        "name": "https"
                    },
                    {
                        "href": "git@bitbucket.org:owner-slug/repostiory-slug.git",
                        "name": "ssh"
                    }
                ],
                "self": {
                    "href": "https://api.bitbucket.org/2.0/repositories/owner-slug/repostiory-slug"
                },
                "source": {
                    "href": "https://api.bitbucket.org/2.0/repositories/owner-slug/repostiory-slug/src"
                },
                "html": {
                    "href": "https://bitbucket.org/owner-slug/repostiory-slug"
                },
                "avatar": {
                    "href": "https://bytebucket.org/ravatar/%7B315caa4f-5259-4ec8-b789-446ae523f770%7D?ts=2081905"
                },
                "hooks": {
                    "href": "https://api.bitbucket.org/2.0/repositories/owner-slug/repostiory-slug/hooks"
                },
                "forks": {
                    "href": "https://api.bitbucket.org/2.0/repositories/owner-slug/repostiory-slug/forks"
                },
                "downloads": {
                    "href": "https://api.bitbucket.org/2.0/repositories/owner-slug/repostiory-slug/downloads"
                },
                "pullrequests": {
                    "href": "https://api.bitbucket.org/2.0/repositories/owner-slug/repostiory-slug/pullrequests"
                }
```

Reply:
Yes, `CloneUrl` must be the http/s (not ssh) url that we use to pull from git. 
`HtmlUrl` might be some cruft here, it might not be needed.

Reply:
Updated the links to use "html" link for HtmlUrl and the "clone[name=https]" link for CloneUrl